lib/srv/app/server_test.go

```diff
@@ -857,19 +872,48 @@ func TestAuthorize(t *testing.T) {
 				RoleAppLabels: test.roleAppLabels,
 			})
 
+			if test.requireTrustedDevice {
+				authServer := s.authServer.AuthServer
+
+				// Create a role that requires a trusted device.
+				requiredDevRole, err := types.NewRole("require-trusted-devices-app", types.RoleSpecV6{
+					Options: types.RoleOptions{
+						DeviceTrustMode: constants.DeviceTrustModeRequired,
+					},
+					Allow: types.RoleConditions{
+						AppLabels: types.Labels{"*": []string{"*"}},
```

Comment:
Does the DeviceTrustMode only apply to resources that the role allows access to?

Whether it does or doesn't, I suppose this will be a breaking change for anyone who may have enabled device trust in the `access` role, for example

Reply:
The breaking change seems fine to me if it goes out in a major

Reply:
Yep, this is for v15.